### PR TITLE
Fixed scoring fe related extension param names

### PR DESCRIPTION
**Pull request by liakaz**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py [hunk 1]
```diff
@@ -189,12 +191,12 @@ def __validate_config(self, configuration_settings, configuration_protected_sett
 
     def __validate_scoring_fe_settings(self, configuration_settings, configuration_protected_settings):
         experimentalCluster = _get_value_from_config_protected_config(
-            'inferenceLoadBalancerHA', configuration_settings, configuration_protected_settings)
-        experimentalCluster = str(experimentalCluster).lower() == 'true'
-        if experimentalCluster:
-            configuration_settings['clusterPurpose'] = 'DevTest'
-        else:
+            self.inferenceLoadBalancerHA, configuration_settings, configuration_protected_settings)
+        notExperimentalCluster = str(experimentalCluster).lower() == 'true'
```
Comment (jonathan-innis) [inline src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py:195]: Something is wrong here, either the variable name or what you are checking, but this statement doesn't quite make sense

File: src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py [hunk 2]
```diff
@@ -206,13 +208,22 @@ def __validate_scoring_fe_settings(self, configuration_settings, configuration_p
...
+        feIsNodePort = _get_value_from_config_protected_config(
+            self.privateEndpointNodeport, configuration_settings, configuration_protected_settings)
+        feIsNodePort = str(feIsNodePort).lower() == 'true'
         feIsInternalLoadBalancer = _get_value_from_config_protected_config(
             self.privateEndpointILB, configuration_settings, configuration_protected_settings)
         feIsInternalLoadBalancer = str(feIsInternalLoadBalancer).lower() == 'true'
-        if feIsInternalLoadBalancer:
+
+        if feIsNodePort and feIsInternalLoadBalancer:
+            raise InvalidArgumentValueError(
```
Comment (jonathan-innis) [inline src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py:219]: This should be a `MutuallyExclusiveArgumentError`

**Review by jonathan-innis — CHANGES_REQUESTED** [review]

File: src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py (hunk 1 above)
Comment (CindyXing) [reply src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py:195]: Lia, is this a typo? It is odd to me too.

File: testing/test/extensions/public/AzureMLKubernetes.Tests.ps1 [hunk 3]
```diff
@@ -107,7 +107,7 @@ Describe 'AzureML Kubernetes Testing' {
     }
 
     It 'Creates the extension and checks that it onboards correctly with inference enabled' {
-        Invoke-Expression "az $Env:K8sExtensionName create -c $($ENVCONFIG.arcClusterName) -g $($ENVCONFIG.resourceGroup) --cluster-type connectedClusters --extension-type $extensionType -n $extensionName --release-train staging --config enableInference=true identity.proxy.remoteEnabled=True identity.proxy.remoteHost=https://master.experiments.azureml-test.net allowInsecureConnections=True inferenceLoadBalancerHA=true" -ErrorVariable badOut
+        Invoke-Expression "az $Env:K8sExtensionName create -c $($ENVCONFIG.arcClusterName) -g $($ENVCONFIG.resourceGroup) --cluster-type connectedClusters --extension-type $extensionType -n $extensionName --release-train staging --config enableInference=true identity.proxy.remoteEnabled=True identity.proxy.remoteHost=https://master.experiments.azureml-test.net allowInsecureConnections=True inferenceLoadBalancerHA=false" -ErrorVariable badOut
```
Comment (CindyXing) [inline testing/test/extensions/public/AzureMLKubernetes.Tests.ps1:110]: Do we have test coverage around privateEndpointNodeport?

File: src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py (hunk 1 above)
Comment (liakaz) [reply src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py:195]: ahhh yes sorry this is a typo. For got to rename experimentalCluster -> notExperimentalCluster in one extra place

Comment (liakaz) [reply src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py]: fixed to a better variable name altogether - isProdCluster

File: testing/test/extensions/public/AzureMLKubernetes.Tests.ps1 (hunk 3 above)
Comment (liakaz) [reply testing/test/extensions/public/AzureMLKubernetes.Tests.ps1:110]: We probably don't. We need to have a discussion about the best places for such tests to go. Will send out an email

Comment (liakaz) [reply testing/test/extensions/public/AzureMLKubernetes.Tests.ps1:110]: For now I just tested manually

**Comment by jonathan-innis** [comment]
@liakaz Can you please resolve the conflicts? Otherwise, this looks good 🚀

**Pull request merged**